### PR TITLE
feat(zk_toolbox): Add prover generate-sk command

### PR DESCRIPTION
**Pull request by matias-gonz**
## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.

### CODE REVIEW

File: zk_toolbox/crates/zk_inception/src/commands/prover/args/init.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+use clap::Parser;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, Parser)]
+pub struct InitArgs {}
```
Comment (matias-gonz) [inline zk_toolbox/crates/zk_inception/src/commands/prover/args/init.rs:5]: Placeholder.
Should we let the user decide where to store the verification keys?

File: zk_toolbox/crates/zk_inception/src/commands/prover/init.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+use xshell::{cmd, Shell};
+
+use super::args::init::InitArgs;
+use crate::messages::{MSG_GENERATING_VK_SPINNER, MSG_INITIALIZING_PROVER, MSG_PROVER_INITIALIZED};
+
+pub(crate) async fn run(_args: InitArgs, shell: &Shell) -> anyhow::Result<()> {
+    logger::info(MSG_INITIALIZING_PROVER);
+    let ecosystem_config = EcosystemConfig::from_file(shell)?;
+    let mut link_to_prover = ecosystem_config.link_to_code.into_os_string();
+    link_to_prover.push("/prover");
```
Comment (aon) [inline zk_toolbox/crates/zk_inception/src/commands/prover/init.rs:13]: This should be stored in the configs crate

File: zk_toolbox/crates/zk_inception/src/commands/ecosystem/create.rs [hunk 3]
```diff
@@ -64,6 +64,9 @@ fn create(args: EcosystemCreateArgs, shell: &Shell) -> anyhow::Result<()> {
         path
     };
 
+    let mut link_to_prover = link_to_code.clone().into_os_string();
+    link_to_prover.push("/prover");
```
Comment (Deniallugo) [inline zk_toolbox/crates/zk_inception/src/commands/ecosystem/create.rs:68]: do we need to store it in config? why not just a function?

File: zk_toolbox/crates/zk_inception/src/commands/prover/args/init.rs (hunk 1 above)
Comment (Deniallugo) [reply zk_toolbox/crates/zk_inception/src/commands/prover/args/init.rs:5]: i think we can start without it and introduce it later

File: zk_toolbox/crates/zk_inception/src/commands/prover/init.rs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    shell.change_dir(&link_to_prover);
+
+    let spinner = Spinner::new(MSG_GENERATING_VK_SPINNER);
+    let mut cmd = Cmd::new(cmd!(
+        shell,
+        "cargo run --features gpu --release --bin key_generator -- 
+            generate-sk all --recompute-if-missing 
+            --setup-path=vk_setup_data_generator_server_fri/data 
+            --path={link_to_prover}/vk_setup_data_generator_server_fri/data"
+    ));
```
Comment (Deniallugo) [inline zk_toolbox/crates/zk_inception/src/commands/prover/init.rs:22]: it's not initializing provers, but more generating keys.

File: ZkStack.yaml [hunk 5]
```diff
@@ -1,6 +1,7 @@
 name: zk
 l1_network: Localhost
 link_to_code: .
+link_to_prover: ./prover
```
Comment (Deniallugo) [inline ZkStack.yaml:4]: seems it's redundant now

File: zk_toolbox/crates/zk_inception/src/commands/prover/generate_sk.rs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    let link_to_prover = get_link_to_prover(&ecosystem_config);
+    shell.change_dir(&link_to_prover);
+
+    let spinner = Spinner::new(MSG_GENERATING_SK_SPINNER);
+    let mut cmd = Cmd::new(cmd!(
+        shell,
+        "cargo run --features gpu --release --bin key_generator -- 
+            generate-sk all --recompute-if-missing 
+            --setup-path=vk_setup_data_generator_server_fri/data 
+            --path={link_to_prover}/vk_setup_data_generator_server_fri/data"
```
Comment (Deniallugo) [inline zk_toolbox/crates/zk_inception/src/commands/prover/generate_sk.rs:20]: should it be absolutize?

Comment (Deniallugo) [reply zk_toolbox/crates/zk_inception/src/commands/prover/generate_sk.rs:20]: i'm just worrying, that we are already in `prover` subdir and we want to go to prover subdir one more time

Comment (matias-gonz) [reply zk_toolbox/crates/zk_inception/src/commands/prover/generate_sk.rs:20]: `link_to_code` is already an absolute path and `link_to_prover` appends `/prover`
Shoud we still use `absolutize()`?

https://github.com/matter-labs/zksync-era/blob/main/zk_toolbox/crates/config/src/ecosystem.rs#L70

**Review by Deniallugo — APPROVED** [review]

**Pull request merged**